web/core/components/core/charts/stacked-bar-chart/root.tsx

```
@@ -0,0 +1,130 @@
+    barSize = 40,
+    className = "w-full h-96",
+    tickCount = {
+      x: undefined,
+      y: 10,
+    },
+    showTooltip = true,
+  }: TStackedBarChartProps<K, T>) => {
+    // derived values
+    const stackKeys = React.useMemo(() => stacks.map((stack) => stack.key), [stacks]);
+    const stackDotClassNames = React.useMemo(
+      () => stacks.reduce((acc, stack) => ({ ...acc, [stack.key]: stack.dotClassName }), {}),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)